# 🛒 Supermarket Sales — EDA Notebook
**Goal:** Load a real dataset, explore it, clean it, and extract insights.

**Dataset:** 50 supermarket transactions across Bangkok, Chiang Mai & Phuket

---

## ✅ Step 1 — Import Libraries

In [2]:
import pandas as pd
import numpy as np

print('Libraries loaded ✅')

Libraries loaded ✅


## ✅ Step 2 — Load the Data

In [3]:
df = pd.read_csv('supermarket_sales.csv')
print(f'✅ Data loaded: {df.shape[0]} rows, {df.shape[1]} columns')

✅ Data loaded: 50 rows, 12 columns


## ✅ Step 3 — First Look
### 3a. head()

In [4]:
df.head()

,order_id,date,category,product,quantity,unit_price,total,customer_age,customer_gender,city,payment_method,rating
0,1001,2024-01-03,Electronics,Headphones,1,59.99,59.99,28,Female,Bangkok,Credit Card,8.5
1,1002,2024-01-04,Food,Chocolate,3,4.50,13.50,35,Male,Chiang Mai,Cash,7.0
2,1003,2024-01-05,Clothing,T-Shirt,2,19.99,39.98,22,Female,Bangkok,E-wallet,9.0
3,1004,2024-01-06,Electronics,Phone Case,1,12.99,12.99,45,Male,Phuket,Credit Card,6.5
4,1005,2024-01-07,Food,Coffee,4,6.75,27.00,30,Female,Bangkok,Cash,8.0


In [5]:
# 📝 YOUR OBSERVATION:
# Each row represents = product types
# Columns I see = order_id, date, category, product, quantity, unit_price, total, customer_age, customer_gender, city, payment_method, rating
# Anything look strange = no

### 3b. shape — How big is the dataset?

In [6]:
print(f'Rows    : {df.shape[0]}')
print(f'Columns : {df.shape[1]}')

Rows    : 50
Columns : 12


### 3c. info() — Data types + missing values

In [7]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 50 entries, 0 to 49
Data columns (total 12 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   order_id         50 non-null     int64  
 1   date             50 non-null     str    
 2   category         50 non-null     str    
 3   product          50 non-null     str    
 4   quantity         50 non-null     int64  
 5   unit_price       50 non-null     float64
 6   total            50 non-null     float64
 7   customer_age     50 non-null     int64  
 8   customer_gender  50 non-null     str    
 9   city             50 non-null     str    
 10  payment_method   50 non-null     str    
 11  rating           50 non-null     float64
dtypes: float64(3), int64(3), str(6)
memory usage: 7.0 KB


In [8]:
# 📝 YOUR OBSERVATION:
# Any missing values? (check non-null count) no
# Is 'date' the right data type? no, because it shows strings
# Any column that looks wrong? no

### 3d. describe() — Statistics summary

In [9]:
df.describe()

,order_id,quantity,unit_price,total,customer_age,rating
count,50.00000,50.000000,50.000000,50.000000,50.000000,50.000000
mean,1025.50000,2.020000,35.322400,41.246800,34.180000,7.870000
std,14.57738,1.285556,53.710521,51.271161,8.735781,1.053711
min,1001.00000,1.000000,1.990000,7.000000,19.000000,6.000000
25%,1013.25000,1.000000,5.312500,14.955000,27.250000,7.000000
50%,1025.50000,2.000000,14.490000,24.475000,33.000000,8.000000
75%,1037.75000,3.000000,43.740000,48.980000,39.750000,8.875000
max,1050.00000,6.000000,299.990000,299.990000,55.000000,9.500000


In [10]:
# 📝 YOUR OBSERVATION:
# Average price = 35.32
# Max total purchase = 299.99
# Average customer rating = 7.87

## ✅ Step 4 — Fix the Date Column

In [11]:
df['date'] = pd.to_datetime(df['date'])
print(f'date column type is now: {df["date"].dtype}')

date column type is now: datetime64[us]


## ✅ Step 5 — Check for Missing Values

In [12]:
print('Missing values per column:')
print(df.isnull().sum())

Missing values per column:
order_id           0
date               0
category           0
product            0
quantity           0
unit_price         0
total              0
customer_age       0
customer_gender    0
city               0
payment_method     0
rating             0
dtype: int64


## ✅ Step 6 — Answer These 3 Questions (Your Turn!)

### ❓ Q1: Which category generates the most total revenue?

In [13]:
# HINT: groupby('category') then sum 'total', then sort
df.groupby('category')['total'].sum().sort_values(ascending=False)
#Electronics

category
Electronics    1007.86
Clothing        644.83
Health          235.82
Food            173.83
Name: total, dtype: float64

### ❓ Q2: Which city has the highest average customer rating?

In [14]:
# HINT: groupby('city') then mean 'rating'
df.groupby('city')['rating'].mean().sort_values(ascending=False)
#Chiang Mai


city
Chiang Mai    8.230769
Bangkok       7.840000
Phuket        7.541667
Name: rating, dtype: float64

### ❓ Q3: What is the most popular payment method?

In [15]:
# HINT: df['payment_method'].value_counts()
df['payment_method'].value_counts().sort_values(ascending=False)
#Credit Card 
#E-wallet


payment_method
Credit Card    17
E-wallet       17
Cash           16
Name: count, dtype: int64

## ✅ Step 7 — Bonus Challenges 🔥
> These separate good students from great ones.

In [16]:
# BONUS 1: Top selling product by quantity?
df.groupby('product')['quantity'].sum().sort_values(ascending=False)
#Protein Bar


product
Protein Bar     6
Rice            5
Snacks          5
Juice           4
Yogurt          4
Socks           4
Coffee          4
Cereal          3
Face Mask       3
Toothbrush      3
Lip Balm        3
Milk            3
Chocolate       3
Bread           3
Hand Cream      2
Cheese          2
Deodorant       2
Eggs            2
Butter          2
Shampoo         2
USB Cable       2
T-Shirt         2
Shorts          2
Pasta           2
Vitamins        2
Green Tea       2
Eye Drops       1
Earbuds         1
Dress           1
Charger         1
Cardigan        1
Body Lotion     1
Belt            1
Hat             1
Scarf           1
Power Bank      1
Mouse           1
Phone Case      1
Jeans           1
Jacket          1
Keyboard        1
Laptop Stand    1
Headphones      1
Speaker         1
Shoes           1
Smart Watch     1
Sunscreen       1
Swimwear        1
Tablet          1
Webcam          1
Name: quantity, dtype: int64

In [17]:
# BONUS 2: Filter only Electronics. What is the average price?
electronics=df[df['category']=='Electronics']
print(electronics['unit_price'].mean())
#76.75


76.75923076923077


In [18]:
# BONUS 3: Which gender spends more on average per transaction?
df.groupby('customer_gender')['total'].mean()
#Female 


customer_gender
Female    42.988846
Male      39.359583
Name: total, dtype: float64

## 📋 Summary — What I Found
> Fill this in after completing the notebook. This becomes your GitHub README.

**Dataset:** 50 supermarket transactions, 3 cities, Jan–Feb 2024

**Key findings:**
1. Electronics dominates revenue at 1,007.86 — nearly double Clothing (644.83)
2. Chiang Mai customers rate highest at 8.23 vs Phuket's 7.54
3. Payment methods are nearly equal — Credit Card and E-wallet tied at 17 uses each

**Tools used:** Python, Pandas

**Author:** (Iung Seangchanmony)

<bound method DataFrame.sum of     order_id   date  category  product  quantity  unit_price  total  \
0      False  False     False    False     False       False  False   
1      False  False     False    False     False       False  False   
2      False  False     False    False     False       False  False   
3      False  False     False    False     False       False  False   
4      False  False     False    False     False       False  False   
5      False  False     False    False     False       False  False   
6      False  False     False    False     False       False  False   
7      False  False     False    False     False       False  False   
8      False  False     False    False     False       False  False   
9      False  False     False    False     False       False  False   
10     False  False     False    False     False       False  False   
11     False  False     False    False     False       False  False   
12     False  False     False    False     Fal